buildkite/build_release.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -xueo pipefail
+trap 'echo build script failed at line $LINENO' ERR
+
+TODAY=$(date +'%Y-%m-%d')
+: ${BRANCH:=master}
+
+if [ ! -d installer ]; then
+    git clone https://github.com/dlang/installer.git installer
+fi
+pushd installer/create_dmd_release
+
+git reset --hard --quiet
+git clean -df --quiet
+if [ $BRANCH != master ] && [ $BRANCH != stable ] &&
+       ! git ls-remote --exit-code --heads origin $BRANCH > /dev/null; then
+    # use master as fallback base for feature branches
+    git fetch origin master --quiet
+else
+    git fetch origin $BRANCH --quiet
+fi
+git checkout FETCH_HEAD --quiet
+
+mkdir build
+LATEST=$(curl -fsSL http://downloads.dlang.org/releases/LATEST)
+rdmd build_all.d v$LATEST $BRANCH && success=yes || success=no
+rm -f build/*.zip # only keep lzma archives
+
+for file in build/*; do
+    gpg --detach-sign --no-use-agent $file
+done
+
+mv build dmd-$BRANCH-$TODAY
+aws --profile ddo s3 sync dmd-$BRANCH-$TODAY/ s3://downloads.dlang.org/nightlies/dmd-$BRANCH-$TODAY/ --acl public-read --cache-control max-age=604800
+popd
+
+if [ ! -d downloads.dlang.org ]; then
+    git clone https://github.com/dlang/downloads.dlang.org.git downloads.dlang.org
+else
+    git -C downloads.dlang.org pull
+fi
+pushd downloads.dlang.org
+if [ ! -f config.json ]; then
+    (umask 0277 && cat > config.json) <<EOS
+{
+    "aws": {
+        "key": "$(sed -n 's|aws_access_key_id\s*=\s*||p' ~/.aws/credentials)",
+        "secret": "$(sed -n 's|aws_secret_access_key\s*=\s*||p' ~/.aws/credentials)",
+        "endpoint": "s3-$(sed -n 's|region\s*=\s||p' ~/.aws/config).amazonaws.com"
+    },
+    "s3bucket": "downloads.dlang.org"
+}
```

Comment:
That looks overly complicated. Why not pass them in simply as environment variables?
(If an attacker gets access to the pipeline he can also `cat` the credentials file)

Reply:
Hardly any way to avoid this, environment variables can be printed as well.
I've added `~/.aws/credentials` parsing in https://github.com/dlang/downloads.dlang.org/pull/5 to reduce the risk of exposing them to the public logs though.